src/main/java/com/lovetropics/gamemodebuild/state/GBPlayerStore.java

```diff
@@ -2,100 +2,138 @@
 
 import com.lovetropics.gamemodebuild.GBConfigs;
 import com.lovetropics.gamemodebuild.GamemodeBuild;
-import net.minecraft.nbt.CompoundTag;
-import net.minecraft.nbt.ListTag;
-import net.minecraft.nbt.Tag;
+import com.lovetropics.gamemodebuild.container.GBStackMarker;
+import com.lovetropics.gamemodebuild.mixin.InventoryAccessor;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.NonNullList;
+import net.minecraft.world.ItemStackWithSlot;
+import net.minecraft.world.entity.EntityEquipment;
+import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public final class GBPlayerStore {
-	private static final String KEY_ACTIVE = "active";
-	private static final String KEY_ENABLED = "enabled";
-	private static final String KEY_PLAYER_INVENTORY = "playerinv";
-	private static final String KEY_BUILD_INVENTORY = "buildinv";
-	private static final String KEY_LIST = "list";
 
 	public static void setEnabled(Player player, boolean enabled) {
-		CompoundTag survivalPlus = getOrCreatePersistent(player, GamemodeBuild.MODID);
-		survivalPlus.putBoolean(KEY_ENABLED, enabled);
+		getOrCreateData(player).enabled = enabled;
+		player.shouldBeSaved();
```

Comment:
This function doesn't have any side-effect, but I'm not sure if it is needed at all to mark the player state dirty - I think(?) it always gets saved